### PR TITLE
TT-1032 implemented way to limit the size of pump records

### PR DESCRIPTION
**Pull request by sredxny**
## Description
Now users will have the ability to limit the size of pump records (in bytes) for all the pumps. Basically we measure and trim the raw Response and Raw Request. They will have to set a value different than zero to the new config `max_record_size`

## Related Issue
https://tyktech.atlassian.net/browse/TT-1032

## Motivation and Context
Provide a solution to https://tyktech.atlassian.net/browse/TT-1032

## How This Has Been Tested
- Ran gw, dash and pump.
- For pump was added in the config file `max_record_size` and set the value 2 and then 1000 (to see how it behaves with greater sizes)
- Send a request to the gw, when `max_record_size` is set to 2 I can see in mongo the RawResponse and RawRequest cut to 2 digits.
- Send a request to the gw, when `max_record_size` is set to 100 I can see in mongo the RawResponse and RawRequest  complete as this data doesn't overpass the 1000 bytes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by sredxny** [comment]
@tbuchaillot  any idea on how to cover it with tests?

File: main.go [hunk 1]
```diff
@@ -200,9 +200,15 @@ func StartPurgeLoop(secInterval int, chunkSize int64, expire time.Duration, omit
 				// Convert to something clean
 				keys := make([]interface{}, len(AnalyticsValues))
 
+				shouldTrim := SystemConfig.MaxRecordSize != 0
 				for i, v := range AnalyticsValues {
 					decoded := analytics.AnalyticsRecord{}
 					err := msgpack.Unmarshal([]byte(v.(string)), &decoded)
+
+					if shouldTrim {
```
Comment (tbuchaillot) [inline main.go:208]: I'd move this if to `FilterData` method.

**Comment by tbuchaillot** [comment]
@sredxny you can test it in `main_test.go` , there are some examples with a mocked pump 👍

**Comment by sredxny** [comment]
@tbuchaillot please check again, I added some changes

**Review by tbuchaillot — APPROVED** [review]
looks good!

**Comment by tbuchaillot** [comment]
@letzya from a DX perspective, what do you think of this? We're basically adding the possibility to trim `raw_request`/ `raw_response` data for all the pumps.

**Review by tbuchaillot — APPROVED** [review]
LGTM

**Pull request merged**